lib/stats/DefaultStatsFactoryPlugin.js

```diff
@@ -98,27 +287,90 @@ const { makePathsRelative, parseResource } = require("../util/identifier");
  * @property {any} _env
  */
 
-/** @template T @typedef {Record<string, (object: Object, data: T, context: UsualContext, options: UsualOptions, factory: StatsFactory) => void>} ExtractorsByOption */
+/**
+ * @template {any} T
+ * @template {Object} O
+ * @typedef {Record<string, (object: O, data: T, context: UsualContext, options: UsualOptions, factory: StatsFactory) => void>} ExtractorsByOption
```

Comment:
Introducing template for `object` so that we can better define the expected output in the usage below.